rust/src/editor.rs

```diff
@@ -530,27 +597,37 @@ impl Editor {
             "render_lines" => Some(self.do_render_lines(params)),
             "key" => async(self.do_key(params)),
             "insert" => async(self.do_insert(params)),
+            "delete_forward" => async(self.delete_forward()),
             "delete_backward" => async(self.delete_backward()),
+            "delete_to_beginning_of_line" => async(self.delete_to_beginning_of_line()),
             "delete_to_end_of_paragraph" => async(self.delete_to_end_of_paragraph(kill_ring)),
             "insert_newline" => async(self.insert_newline()),
-            "move_up" => async(self.move_up(0)),
-            "move_up_and_modify_selection" => async(self.move_up(MODIFIER_SHIFT)),
```

Comment:
I was thinking of keeping an argument, just having it a bool for `modify_selection`. So `move_up` would call `self.move_up(false)` and `move_up_and_modify_selection` would call `self.move_up(true)`. In general, I like less statefulness and more of a functional style.